### ISSUE TITLE
If verification takes too long, it results in an ExecutionException

### ISSUE THREAD

**Issue by Sophietje** [utwente-fmt/vercors#426]
I've been working on verifying selection sort. However, I've now arrived at a point where the verification takes too long. As a result, Vercors terminates with an exception:
```
[removed some debug information here]
Verification aborted exceptionally: java.util.concurrent.ExecutionException: java.util.concurrent.ExecutionException: java.lang.NullPointerException
  Cause: java.util.concurrent.ExecutionException: java.lang.NullPointerException
Errors! (1)
[warning] unspecified error exceptional.error
[progress] task Viper verification took 3978765 ms
[progress]  ... pass took 423 ms
[debug] At Main.java:669:
The final verdict is Fail
[progress] entire run took 4011135 ms
```
It would be nicer to let the user know that the verification took too long and the process has therefore been terminated (I'm assuming that's what happened) rather than showing an exception. Moreover, this output suggests that Viper did manage to (dis)prove it at some point since it mentions the "Viper verification took XX ms".

Code for reproduction of error:
```
// Code taken (and modified) from: https://www.geeksforgeeks.org/selection-sort/
// Verified the following properties:
// - The algorithm results in arr being a sorted list
// - Resulting lists have the same number of elements

// Still need to prove:
// - The algorithm results in arr being a permutation of the original input array

class SelectionSort {
	
	context xs != null && xs.length > 0 && fromIndex >= 0;
	requires (\forall* int j; j >= 0 && j < xs.length; Perm(xs[j], write));
	ensures (\forall int j; j >= fromIndex && j < xs.length; xs[j] in \result > 0);
	ensures |\result| == xs.length - fromIndex;
	pure bag<int> toBagFrom(int[] xs, int fromIndex) = fromIndex < xs.length ? bag<int>{xs[fromIndex]} + toBagFrom(xs, fromIndex+1) : bag<int> {};


	context arr != null;
	context arr.length > 0;
	context (\forall* int j; j >= 0 && j < arr.length; Perm(arr[j], write));
	ensures arr.length == \old(arr.length);
	ensures (\forall int j; j >= 0 && j < arr.length-1; arr[j] <= arr[j+1]);
	ensures toBagFrom(arr, 0) == toBagFrom(\old(arr), 0);
        void sort(int[] arr) { 
            int n = arr.length; 

	    loop_invariant (\forall* int j; j >= 0 && j < arr.length; Perm(arr[j], write));
            loop_invariant i >= 0;
	    loop_invariant n == arr.length && arr.length > 0;
	    loop_invariant i <= n-1;
	    loop_invariant (\forall int j; j >= 0 && j < i; (\forall int k; k > j && k < n; arr[j] <= arr[k]));
	    loop_invariant (\forall int j; j >= 0 && j < i; arr[j] <= arr[j+1]);
	    loop_invariant toBagFrom(arr, 0) == toBagFrom(\old(arr), 0);
            // One by one move boundary of unsorted subarray 
            for (int i = 0; i < n-1; i=i+1) { 
			
            // Find the minimum element in unsorted array 
            int min_idx = i; 
			
	        loop_invariant min_idx >= i && min_idx < n;
		loop_invariant j >= i && j <= n;
		loop_invariant i >= 0;
		loop_invariant i <= n-1;
		loop_invariant n == arr.length && arr.length > 0;
		loop_invariant (\forall* int k; k >= 0 && k < arr.length; Perm(arr[k], write));
		loop_invariant (\forall int k; k >= 0 && k < i-1; arr[k] <= arr[k+1]);
		loop_invariant (\forall int k; k >= i && k < j; arr[min_idx] <= arr[k]);
		loop_invariant (\forall int k; k >= 0 && k < i; arr[k] <= arr[min_idx]);
		loop_invariant (\forall int j; j >= 0 && j < i; (\forall int k; k > j && k < n; arr[j] <= arr[k]));
		loop_invariant toBagFrom(arr, 0) == toBagFrom(\old(arr), 0);
                for (int j = i+1; j < n; j=j+1) {
	            assert j >= 0 && j < arr.length;
                    if (arr[j] < arr[min_idx]) {
                        min_idx = j; 
			assert min_idx >= i && min_idx < n;
		    }
            }

            // Swap the found minimum element with the first element 
            swap(arr, min_idx, i);
        } 
    } 
	
	
	context arr != null && arr.length > 0;
	context min_idx >= 0 && min_idx < arr.length;
	context i >= 0 && i < arr.length;
	context (\forall* int j; j >= 0 && j < arr.length; Perm(arr[j], write));
	ensures (\forall int j; j >= 0 && j < arr.length && j != min_idx && j != i; \old(arr[j]) == arr[j]);
	ensures arr[min_idx] == \old(arr[i]);
	ensures arr[i] == \old(arr[min_idx]);
	ensures toBagFrom(arr, 0) == toBagFrom(\old(arr), 0);
	void swap(int[] arr, int min_idx, int i) {
		int temp = arr[min_idx];
		arr[min_idx] = arr[i];
		arr[i] = temp;
	}
 
    void main() { 
        SelectionSort ob = new SelectionSort(); 
        int[] arr = new int[5];
	arr[0] = 64; arr[1] = 25; arr[2] = 12; arr[3] = 22; arr[4] = 11;
        ob.sort(arr);
    } 
} 
```

**Comment by bobismijnnaam** [comment]
In my experience a concurrent ExecutionException is usually a viper error that we're not handling properly. For example, I ran into a few instances where vercors produced code that unfolds an abstract predicate. That's not possible, so viper errors, but vercors reports it as as an exception for some reason. So it could be a timeout error or something comparable in your case.

I tried verifying the silver output of your test case both in vscode and in the newest version of silicon. vscode verifies it without problems, and the newest version takes a long time (I haven't seen it terminate yet). So it seems indeed viper has made some changes.
